### PR TITLE
Shuttles Episode IV: A New Hope NOT WIP, READY FOR TESTING

### PR DESCRIPTION
**Pull request by spookydonut**
It is a period of civil war. 
New shuttles, striking from a hidden codebase, have won their first victory against the evil Shitcode Empire.
During the battle, goodcoder spies managed to steal secret plans to the Empire's ultimate weapon, the old shuttlecode, an armored kernel of shitcode with enough power to destroy an entire round.  
Pursued by the Empire's sinister agents, spookydonut races home aboard their new shuttles, custodian of the stolen plans that can save their fellow coders and restore lagfree gaming to the codebase.....

Ports last part of the 4 part shuttle rework/port

:cl:
add: Ports dropships to new shuttle system
del: Removed last of old shuttle stuff
tweak: Queen now has a verb for calling down the shuttle, it will call it down to the closest LZ to the queen no matter what the shuttle is doing, can't calldown before 30 minutes or if more than 10% alive humans are on the ground.
del: Removed normandy for now
add: The Alamo can land at either LZ and fly between LZs
del: Removed CAS for now
imageadd: half tile shuttle pieces now overlay properly on turfs, no more plating welded to the dropship as it flies.
/:cl:

### CODE REVIEW

**Comment by Ghommie** [comment]
11/10 on the PR description. Good job overall.

**Comment by Pariah919** [comment]
Now this is some next level shitposting.

**Comment by Jalleo** [comment]
Spoiler going to end up with revenge of the shitcode.

**Comment by adamkad1** [comment]
Or the return of the goodcoders

**Comment by Jalleo** [comment]
No thats after

File: code/game/machinery/computer/dropship_weapons.dm [hunk 1]
```diff
@@ -25,7 +25,7 @@
 
 
 /obj/machinery/computer/dropship_weapons/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 0)
-	var/data[0]
+/*	var/data[0]
```
Comment (DominikPanic) [inline code/game/machinery/computer/dropship_weapons.dm]: Commented out code.

File: code/game/machinery/door_control.dm [hunk 2]
```diff
@@ -25,21 +25,21 @@
...
-				shuttle_tag = "[CONFIG_GET(string/ship_name)] Dropship 2"
-			else
-				return
-
-		var/datum/shuttle/ferry/marine/shuttle = shuttle_controller.shuttles[shuttle_tag]
-		shuttle.hijack(M)
-		shuttle.door_override(M)
-	else
-		..()
+//	if(M.xeno_caste.caste_flags & CASTE_IS_INTELLIGENT && normaldoorcontrol == CONTROL_DROPSHIP)
```
Comment (DominikPanic) [inline code/game/machinery/door_control.dm]: Commented out code

File: code/game/machinery/door_control.dm [hunk 3]
```diff
@@ -51,7 +51,7 @@
 	return src.attack_hand(user)
 
 /obj/machinery/door_control/proc/handle_dropship(var/ship_id)
-	var/shuttle_tag
+/*	var/shuttle_tag
```
Comment (DominikPanic) [inline code/game/machinery/door_control.dm]: Commented out code

File: code/game/machinery/door_control.dm [hunk 4]
```diff
@@ -132,21 +132,21 @@
 				if(specialfunctions & SAFE)
 					D.safe = 1
 
-/obj/machinery/door_control/proc/handle_pod()
+/obj/machinery/door_control/proc/handle_pod()/*
```
Comment (DominikPanic) [inline code/game/machinery/door_control.dm]: Commented out code

File: code/modules/cm_marines/dropship_equipment.dm [hunk 5]
```diff
@@ -47,11 +47,11 @@
 			installed_equipment = SE
 			SE.ship_base = src
 
-			for(var/datum/shuttle/ferry/marine/S in shuttle_controller.process_shuttles)
+/*			for(var/datum/shuttle/ferry/marine/S in shuttle_controller.process_shuttles)
```
Comment (DominikPanic) [inline code/modules/cm_marines/dropship_equipment.dm]: Commented out code

**Comment by spookydonut** [comment]
> 
> 
> Plus there's a lot more, I assume some is going to be replaced, but could you remove the one that is not being used?

getting to it, theres still a bunch of cleanup to do and queen calldown to do.
oh and admin tools but the basic functionality is done and people other than me should start testing it.

File: code/game/machinery/door_control.dm (hunk 4 above)
Comment (Rohesie) [reply code/game/machinery/door_control.dm]: This is breaking all shutter control.

**Comment by Rohesie** [comment]
Check the comment on the change request.

File: code/game/machinery/door_control.dm (hunk 4 above)
Comment (spookydonut) [reply code/game/machinery/door_control.dm]: handle_pod isnt used

**Comment by Rohesie** [comment]
Let us know when it's testable.

**Comment by spookydonut** [comment]
> 
> 
> Let us know when it's testable.

I did, and then removed the WIP tag

**Comment by psykzz** [comment]
Ran a test locally with 4 people
## Setup:
Starting with 1 queen, 2 humans (1 observer)
1 human planetside, varedit round start time to some low number to pass 30 mins mark

## Test case
* Queen tried to call down, "error 50% humans on planet."
* Queen kills human
* Queen tried to call down, "Something went wrong."
* Alamo leaves and arrived at Lz2
* Dropship popup allows sending ship but just resets the timer.

image

**Comment by Rohesie** [comment]
Needs the issue fixed before we can test it again.

**Pull request merged**